### PR TITLE
feat(routes): add 南京晓庄学院图书馆通知公告、江苏海洋大学通知公告、江苏海洋大学研招网通知公告

### PR DESCRIPTION
**Pull request by real-jiakai**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/njxzc/libtzgg
/jou/tzgg
/jou/yztzgg
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

添加南京晓庄学院图书馆通知公告rss、江苏海洋大学通知公告rss、江苏海洋大学研招网通知公告rss的支持。

### CODE REVIEW

File: lib/v2/njxzc/maintainer.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
 module.exports = {
     '/tzgg': ['real-jiakai'],
+    '/libtzgg': ['real-jiakai'],
```
Comment (TonyRL) [inline lib/v2/njxzc/maintainer.js:3]:
```suggestion
    '/libtzgg': ['real-jiakai'],
    '/tzgg': ['real-jiakai'],
```
Please insert new entries in alphabetical order

File: lib/v2/njxzc/radar.js [hunk 2]
```diff
@@ -9,5 +9,13 @@ module.exports = {
                 target: '/njxzc/tzgg',
             },
         ],
+        lib: [
```
Comment (TonyRL) [inline lib/v2/njxzc/radar.js:12]: Please insert new entries in alphabetical order

File: lib/v2/njxzc/router.js [hunk 3]
```diff
@@ -1,4 +1,6 @@
 module.exports = (router) => {
-    // 南京晓庄学院官网通知公告
+    // 南京晓庄学院通知公告
     router.get('/tzgg', require('./home'));
+    // 南京晓庄学院图书馆通知公告
+    router.get('/libtzgg', require('./lib'));
```
Comment (TonyRL) [inline lib/v2/njxzc/router.js:5]:
```suggestion
    // 南京晓庄学院图书馆通知公告
    router.get('/libtzgg', require('./lib'));
    // 南京晓庄学院通知公告
    router.get('/tzgg', require('./home'));
```
Please insert new entries in alphabetical order

File: docs/university.md [hunk 4]
```diff
@@ -1650,6 +1650,16 @@ jsjxy.hbut.edu.cn 证书链不全，自建 RSSHub 可设置环境变量 NODE_TLS
 
 </Route>
 
+## 江苏海洋大学
+
+<Route author="real-jiakai" example="/jou/tzgg" path="/jou/:type" :paramsDesc="分类名" radar="1">
```
[the remark this answers is not in the document]
Comment (real-jiakai) [reply docs/university.md:1655]: Ok. I will add h3 heading later.

**Comment by real-jiakai** [comment]
I have updated codes. Pls check it when free.

File: docs/university.md [hunk 5]
```diff
@@ -1650,6 +1650,16 @@ jsjxy.hbut.edu.cn 证书链不全，自建 RSSHub 可设置环境变量 NODE_TLS
 
 </Route>
 
+## 江苏海洋大学
```
Comment (Supertianw) [inline docs/university.md:1653]: 氨水

File: lib/v2/jou/utils/index.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        .toArray()
+        .map((item) => {
+            const currentItem = $(item); // 获取当前tr元素的jQuery对象
+            const item1 = currentItem.find('td:eq(1)'); // 获取当前tr元素下的第二个td
+            const item2 = currentItem.find('td:eq(2)'); // 获取当前tr元素下的第三个td
+            const link = host + item1.find('a').attr('href');
+
+            return {
+                title: item1.find('a').attr('title'),
+                link: link.replace('..', ''),
```
Comment (TonyRL) [inline lib/v2/jou/utils/index.js:34]: Please fix CodeQL failure
```suggestion
            const link = new URL(item1.find('a').attr('href'), url).href;

            return {
                title: item1.find('a').attr('title'),
                link,
```

File: lib/v2/jou/utils/index.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    // 通知公告的items的标题、url链接、发布日期
+    const list = $(`table.${tableClass} > tbody > tr`)
+        .filter((index, item) => {
+            const currentItem = $(item);
+            const item1 = currentItem.find('td:eq(1)');
+            const hasLink = item1.find('a').length > 0;
+            const isLastItem = index === $(`table.${tableClass} > tbody > tr`).length - 1;
+            return hasLink && !isLastItem;
+        })
+        .toArray()
```
Comment (TonyRL) [inline lib/v2/jou/utils/index.js:25]: If you want to remove the last item from an array, you can use 
```js
.slice(0, -1)
```

You can also filter at the selector level
```suggestion
    const list = $(`table.${tableClass} > tbody > tr[height=20]`)
        .toArray()
```

File: lib/v2/njxzc/radar.js [hunk 8]
```diff
@@ -1,10 +1,18 @@
...
+                docs: 'https://docs.rsshub.app/university.html#nan-jing-xiao-zhuang-xue-yuan-guan-wang-tong-zhi-gong-gao',
+                source: ['/pxyhd/list.htm', '/'],
+                target: '/njxzc/libtzgg',
+            },
+        ],
         www: [
             {
                 title: '通知公告',
-                docs: 'https://docs.rsshub.app/university.html#nan-jing-xiao-zhuang-xue-yuan-tong-zhi-gong-gao',
+                docs: 'https://docs.rsshub.app/university.html#nan-jing-xiao-zhuang-xue-yuan-tu-shu-guan--tong-zhi-gong-gao',
```
Comment (TonyRL) [inline lib/v2/njxzc/radar.js:15]:
```suggestion
                docs: 'https://docs.rsshub.app/university.html#nan-jing-xiao-zhuang-xue-yuan-tong-zhi-gong-gao',
```

File: lib/v2/jou/utils/index.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+        .map((item) => {
+            const currentItem = $(item); // 获取当前tr元素的jQuery对象
+            const item1 = currentItem.find('td:eq(1)'); // 获取当前tr元素下的第二个td
+            const item2 = currentItem.find('td:eq(2)'); // 获取当前tr元素下的第三个td
+            const link = new URL(item1.find('a').attr('href'), host).href;
+
+            return {
+                title: item1.find('a').attr('title'),
+                link,
+                pubDate: timezone(parseDate(item2.find(`.${timeStyleClass}`).text(), 'YYYY-MM-DD'), +8),
```
Comment (TonyRL) [inline lib/v2/jou/utils/index.js:28]: Invalid Date in https://github.com/DIYgod/RSSHub/pull/12176#issuecomment-1484077713 and https://github.com/DIYgod/RSSHub/pull/12176#issuecomment-1484077716
The class name of these two isn't the same.

File: lib/v2/jou/utils/index.js [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+                    item.description = '该通知无法直接预览，请点击原文链接↑查看';
+                } else {
+                    const $ = cheerio.load(response.data);
+                    item.title = $(`.${titleStyleClass}`).text();
+                    item.description = $('.v_news_content')
+                        .html()
+                        .replace(/src="\//g, `src="${new URL('.', host).href}`)
+                        .replace(/href="\//g, `href="${new URL('.', host).href}`)
+                        .trim();
+                    item.pubDate = timezone(parseDate($(`.${timeStyleClass}`).text().replace('发布时间：', '')), +8);
```
Comment (TonyRL) [inline lib/v2/jou/utils/index.js:48]: Invalid Date in https://github.com/DIYgod/RSSHub/pull/12176#issuecomment-1484077713 and https://github.com/DIYgod/RSSHub/pull/12176#issuecomment-1484077716
The class name of these two isn't the same.

File: lib/v2/njxzc/utils/index.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+                    item.description = '该通知无法直接预览，请点击原文链接↑查看';
+                } else {
+                    const $ = cheerio.load(response.data);
+                    item.title = $(contentSelector.title).text();
+                    item.description = $(contentSelector.content)
+                        .html()
+                        .replace(/src="\//g, `src="${new URL('.', host).href}`)
+                        .replace(/href="\//g, `href="${new URL('.', host).href}`)
+                        .trim();
+                    item.pubDate = timezone(parseDate($(contentSelector.date).text().replace('发布时间：', '')), +8);
```
Comment (TonyRL) [inline lib/v2/njxzc/utils/index.js:36]: Invalid date in https://github.com/DIYgod/RSSHub/pull/12176#issuecomment-1484077707
```suggestion
                    item.pubDate = timezone(parseDate($(contentSelector.date).text().replace('编辑：', '').replace('发布日期：', '')), +8);
```

**Pull request merged**